ovirt/resource_ovirt_vm.go

```diff
@@ -376,6 +382,71 @@ func resourceOvirtVMCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	vmBuilder.Cpu(cpu)
 
+	if storage_domain, _ := blockDevice.([]interface{})[0].(map[string]interface{})["storage_domain"]; storage_domain != "" && templateIDOK {
```

Comment:
this code will panic if there is no blockDevice. must check for a block device existence.
use `blockDeviceOK` which is set on line 316

Reply:
This code will not cause panic because in line 334 we already checked that block device is OK
From line 334:
```
		if len(tds) == 0 && !blockDeviceOk {
			return fmt.Errorf("template has no disks attached, so block_device must be assigned")
		}
```

Reply:
`panic: runtime error: index out of range [0] with length 0`

Yup, the code panics if a block device isn't defined. This is in the situation where a template is defined, but block device isn't.

As @rgolangh says, wrapping that whole if inside `if blockDeviceOk` solves the issue.

Reply:
blockDebiceOk check was added - https://github.com/oVirt/terraform-provider-ovirt/pull/216/commits/f67b88faf8248f1db29dc104a345d7a97b0068b0